src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -99,8 +99,7 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
 
     if (mSystemState != nullptr)
     {
-        mSystemState->Release();
-        chip::Platform::Delete(mSystemState);
+        Platform::Delete(mSystemState);
```

Comment:
So this is kind of weird: this is explicitly deleting a reference-counted object.... I guess the assumption is that consumers will have all released their refs and the DeviceControllerFactory does not actually hold a ref?

Reply:
Well, the factory was deleting this regardless of the refcount from the start.

This is less weird to me than the original design, which was to Shutdown when there is 1-ref remaining and Delete when there are 0 (and then free again in the factory). I do not think Release() should do anything unless the count becomes zero.

In a previous version of this fix, I kept the initial refcount as 1 and the self-delete, but at the time I didn't realize it was as weird as it was and the Mac tests were failing. Now I tried this way, and it feels less weird to me, but the tests are still failing :(